### PR TITLE
Allocation Receipts

### PR DESCRIPTION
**Pull request by bh2smith**
A quick script that accepts an account file as input, opens up all the relevant data used to build the final allocation and prints an "allocation receipt"

# Test Plan

Requires having previously generated all the output data (by running main). 

Create an `accounts.csv` file for the accounts you'd like receipts. Or use the sample one provided here

```
python -m scripts.allocation_receipts
```

Should result in output like this:

```
==========================================
Account
0x1e9abb8087f733ef6ab7a0e834bf0e3164fe36be
------------------------------------------
GNO Holder Details
  mainnet:           2.991
  gnosis chain:      0.000
  total gno:         2.991
  holder allocation: 94.470
------------------------------------------
Trader Details
  eligible volume:   1731
  num trades:        6
  first trade date:  2021-05-17
  last trade date:   2021-12-31
  days between:      228
------------------------------------------
  allocation type:   Primary
  trader allocation: 995.160
==========================================
==========================================
Account
0x262d23a2d916f6cf08e0235315aa51e22d142d0b
------------------------------------------
GNO Holder Details
  mainnet:           5.453
  gnosis chain:      1.952
  total gno:         7.405
  holder allocation: 233.843
------------------------------------------
Trader Details
  eligible volume:   192775
  num trades:        193
  first trade date:  2021-04-09
  last trade date:   2022-01-07
  days between:      273
------------------------------------------
  allocation type:   Primary
  trader allocation: 7961.279
==========================================
==========================================
Account
0xaa92521387df9cf483f0c3e0e6f5424e9998316f
------------------------------------------
GNO Holder Details
  Total GNO balance < 0.1
------------------------------------------
Trader Details
  eligible volume:   51045
  num trades:        1
  first trade date:  2021-11-17
  last trade date:   2021-11-17
  days between:      0Trading Criteria Met
  volume: True
  trades: False
  days:   False
------------------------------------------
  allocation type:   Consolation
  trader allocation: 99.807
==========================================
==========================================
Account
0xbeef4dade5ca40347ed018bf80c6fd244ed44492
------------------------------------------
GNO Holder Details
  Total GNO balance < 0.1
------------------------------------------
Trader Details
  No trading activity prior to snapshot
==========================================
==========================================
Account
0xd9c2624b735c6ded3cd0bb79f04feb038cf73aa9
------------------------------------------
GNO Holder Details
  Total GNO balance < 0.1
------------------------------------------
Trader Details
  No trading activity prior to snapshot
==========================================
```

### CODE REVIEW

File: src/scripts/allocation_receipts.py [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+        type=str,
+        help="file containing ethereum addresses",
+    )
+    args = parser.parse_args()
+    # Load all Data related to allocations and where they were derived from
+
+    allocation_details = load_all_data_from_out(AllocationFiles())
+
+    accounts = sorted(
+        Account.load_from(File(name=args.account_file, path='./out/')),
```
Comment (fleupold) [inline src/scripts/allocation_receipts.py]: I found it counterintuitive that this file is expected in a certain path. If it's passed in as a commandline argument, I'd expect it to be relative to the folder from which I execute the command.

File: src/scripts/allocation_receipts.py [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+        key=lambda t: t.account
+    )
+
+    receipts = [allocation_details.account_detail_string(a.account) for a in accounts]
+    with open(File('allocation-receipts.csv').filename(), 'w', encoding='utf-8') as out:
+        out.write("\n".join(receipts))
```
Comment (fleupold) [inline src/scripts/allocation_receipts.py]: Have you considered if printing is maybe more useful for the user of the script (it's trivial for a user to pipe the output into a file)?

**Review by fleupold — APPROVED** [review]
Very useful scrip! Code looks good. Regarding the test

> Or use the sample one provided here

I wasn't able to find that file in the PR.

> python -m scripts.allocation_receipts

I had to run `python3 -m **src**.scripts.allocation_receipts **--account-file accounts.csv** `

**Comment by bh2smith** [comment]
I deleted the sample file from the project because I dont think it's worth keeping around.

File: src/scripts/allocation_receipts.py (hunk 1 above)
Comment (bh2smith) [reply src/scripts/allocation_receipts.py]: In that case the user is going to have to provide both the path and the file name as a runtime argument. I can probably split the full path for them on the last / and deal with it internally.

File: src/scripts/allocation_receipts.py (hunk 2 above)
Comment (bh2smith) [reply src/scripts/allocation_receipts.py]: I was printing the results in a commit just before the most recent commit, but piping the results to file is a great suggestion. I think I may do that.

**Pull request merged**